projects/ngx-flamegraph/src/lib/flamegraph/flamegraph.component.ts

```diff
@@ -152,4 +167,39 @@ export class FlamegraphComponent implements OnInit, OnDestroy {
     const idx = parseInt(dataIdx, 10);
     return this.entries[idx] ?? null;
   }
+
+  private _showTooltip(entry: Data, startCoor: Coor) {
+    if (!this.tooltipEnabled || this.getWidth(entry) > entry.label.length * LETTER_WIDTH) {
```

Comment:
I want to point out that this isn't an ideal solution but rather a cheap way to determine if the text is truncated. The other alternative is to render the text in a container and measure the occupied width. This, however, will lead to many layout recalculations which we don't want for such a minor feature. Another option is to create an algorithm that guesses the width based on font size which is not ideal either. 

The `LETTER_WIDTH` is set to `14` which should be pretty pessimistic, so the tooltip will be shown even when the container width is wide enough to fit the full text.